File: perfect.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool isPerfectNumber(int number) {
    int sum = 1;
    for (int i = 2; i <= sqrt(number); i++) {
        if (number % i == 0) {
            sum += i + number / i;
        }
    }
    return sum == number;
}

int main() {
    int number;
    cout << "Enter a number: ";
    cin >> number;
    if (isPerfectNumber(number)) {
        cout << number << " is a perfect number." << endl;
    } else {
        cout << number << " is not a perfect number." << endl;
    }
    return 0;
}

File: perfect.js
let num = 28;
let sum = 0;

for (let i = 1; i < num; i++) {
    if (num % i == 0) {
        sum += i;
    }
}

if (sum == num) {
    console.log(num + " is a perfect number.");
} else {
    console.log(num + " is not a perfect number.");
}
